### PR TITLE
fix makefile for macOS Sequoia & Ubuntu 24.04.01

### PR DESCRIPTION
**Pull request by efremropelato**
- macOS Sequoia 15.0.1 = error invalid signature
Add flags `-ldflags -s` to build as (https://github.com/golang/go/issues/19734)

> error after build without new flags:
> Screenshot 2024-10-28 alle 13 17 18

- Ubuntu 24.04.01 LTS = env XDG_DATA_DIRS is not empty
Generalized the definition of the JR_SYSTEM_DIR env variable
> Schermata del 2024-10-28 13-20-18

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -67,7 +46,7 @@ compile:
 	-X 'github.com/jrnd-io/jr/pkg/cmd.GoVersion=$(GOVERSION)' \
 	-X 'github.com/jrnd-io/jr/pkg/cmd.BuildUser=$(USER)' \
 	-X 'github.com/jrnd-io/jr/pkg/cmd.BuildTime=$(TIME)'" \
-	-o build/jr jr.go
+	-ldflags -s -o build/jr jr.go
```
Comment (vmarchese) [inline Makefile:49]: that's ok. Please add also the `-w` option to also strip DWARF symbols

**Comment by vmarchese** [comment]
Can you please submit two different PRs? One for the ldflags option and another for the env vars detection

**Comment by efremropelato** [comment]
I think to do it, it would be best to reject this PR. 
I’ll then create two separate PRs to address the different changes. ok?

**Comment by efremropelato** [comment]
OK, I will close this PR

**Pull request closed without merging**